authentication/pipeline/user.py

```diff
@@ -87,17 +88,54 @@ def create_user_via_email(
     data["email"] = kwargs.get("email", kwargs.get("details", {}).get("email"))
 
     if "name" not in data or "password" not in data:
-        raise RequirePasswordAndProfileException(backend, current_partial)
+        raise RequirePasswordAndAddressException(backend, current_partial)
 
     serializer = UserSerializer(data=data)
 
     if not serializer.is_valid():
-        raise RequirePasswordAndProfileException(
+        raise RequirePasswordAndAddressException(
             backend, current_partial, errors=serializer.errors
         )
     return {"is_new": True, "user": serializer.save()}
 
 
+@partial
+def create_profile_via_email(
+    strategy, backend, user=None, flow=None, current_partial=None, *args, **kwargs
+):  # pylint: disable=too-many-arguments,unused-argument
+    """
+    Creates a new profile for the user
+    Args:
+        strategy (social_django.strategy.DjangoStrategy): the strategy used to authenticate
+        backend (social_core.backends.base.BaseAuth): the backend being used to authenticate
+        user (User): the current user
+        flow (str): the type of flow (login or register)
+        current_partial (Partial): the partial for the step in the pipeline
+
+    Raises:
+        RequireProfileException: if the profile data is missing
+    """
+    if backend.name != EmailAuth.name or flow != SocialAuthState.FLOW_REGISTER:
+        return {}
+
+    data = strategy.request_data().copy()
+    if "birth_year" not in data:
```

Comment:
Would this not be covered in the `is_valid()` check for the `ProfileSerializer`? If not, should it be?

Reply:
should be covered in ProfileSerializer